rocAL/rocAL/source/coco_meta_data_reader_key_points.cpp

```diff
@@ -0,0 +1,388 @@
+                    }
+                    else if (0 == std::strcmp(internal_key, "is_crowd"))
+                    {
+                        is_crowd = parser.GetInt();
+                    }
+                    else if (0 == std::strcmp(internal_key, "area"))
+                    {
+                        area = parser.GetDouble();
+                    }
+                    else if (0 == std::strcmp(internal_key, "bbox"))
+                    {
+                        RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+                        parser.EnterArray();
+
+                        box_center.push_back(parser.NextArrayValue() * parser.GetDouble());
+                        box_center.push_back(parser.NextArrayValue() * parser.GetDouble());
+                        box_scale.push_back(parser.NextArrayValue() * parser.GetDouble());
+                        box_scale.push_back(parser.NextArrayValue() * parser.GetDouble());
+
+                        // Move to next section
+                        parser.NextArrayValue();
+                    }
+                    else if (0 == std::strcmp(internal_key, "keypoints"))
+                    {
+                        RAPIDJSON_ASSERT(parser.PeekType() == kArrayType);
+                        parser.EnterArray();
+                        int i = 0;
+                        while (parser.NextArrayValue())
+                        {
+                            keypoint[i] = parser.GetDouble();
+                            joint_sum += keypoint[i];
+                            ++i;
+                        }
+                    }
+                    else
+                    {
+                        parser.SkipValue();
+                    }
+                }
+                char buffer[13];
+                sprintf(buffer, "%012d", id);
+                string str(buffer);
+                std::string file_name = str + ".jpg";
+                auto it = _map_img_sizes.find(file_name);
+                ImgSizes image_size = it->second; // Normalizing the co-ordinates & convert to "ltrb" format
+
+                // Ignore annotations if
+                // label is not person (label !=1)
+                // joint_sum <= 0
+                // is_crowd==1
+                if (label != 1 || joint_sum <= 0 || is_crowd == 1)
+                {
+                    box_center.clear();
+                    box_scale.clear();
+                    continue;
+                }
+
+                // Validate bbox values
+                float x1, y1, x2, y2;
+                x1 = (box_center[0] > 0) ? box_center[0] : 0;
+                y1 = (box_center[1] > 0) ? box_center[1] : 0;
+                float box_w = ((box_scale[0] - 1) > 0) ? (box_scale[0] - 1) : 0;
+                float box_h = ((box_scale[1] - 1) > 0) ? (box_scale[1] - 1) : 0;
+                x2 = ((image_size[0].w - 1) < (x1 + box_w)) ? (image_size[0].w - 1) : (x1 + box_w);
```

Comment:
avoid this condition and use std::max for clamping (x2, y2) to boudary

Reply:
Done